### PR TITLE
Pure Python connector to remove MySQL binary dep

### PR DESCRIPTION
**Pull request by jrs65**
- refactor: use pure python connector that doesn't require mysql libraries
- refactor: remove Python 2 patches

### CODE REVIEW

File: chimedb/core/connectdb.py [hunk 1]
```diff
@@ -262,7 +237,7 @@ def execute_sql(self, sql, params=None, commit=True):
         return cursor
 
 
-class MySQLDatabaseReconnect(RetryOperationalError, pw.MySQLDatabase):
+class MySQLDatabaseReconnect(RetryOperationalError, MySQLConnectorDatabase):
```
Comment (ketiltrout) [inline chimedb/core/connectdb.py:240]: What's up with this?

**Comment by ketiltrout** [comment]
If you're removing the Python-2 work-arounds, do you also want to remove this: https://github.com/chime-experiment/chimedb/blob/8874b08aaef59123991fc75831f4a954d8c4acbd/chimedb/core/context.py#L63

Not sure what we would gain, though.

File: chimedb/core/connectdb.py (hunk 1 above)
Comment (jrs65) [reply chimedb/core/connectdb.py:240]: I'm not following. What's wrong with it?

Comment (ketiltrout) [reply chimedb/core/connectdb.py:240]: Sorry.  Why are you changing it?  Was it a Py2 thing?

Comment (jrs65) [reply chimedb/core/connectdb.py:240]: No. It's switching peewee's backend to the pure python connector (i.e. the actual point of this PR).

Comment (ketiltrout) [reply chimedb/core/connectdb.py:240]: ahhhhhhhhhhhhhhhhhh

**Review by ketiltrout — APPROVED** [review]

**Pull request merged**